content/news/2022-07-29-bevy-0.8/index.md

```diff
@@ -0,0 +1,1131 @@
+    .with_children(|parent| {
+        parent.spawn_bundle(SpriteBundle {
+            texture: player_texture,
+            ..default()
+        });
+        parent.spawn_bundle(SpriteBundle {
+            texture: hat_texture,
+            ..default()
+        });
+    });
+```
+
+However for logic that adds/removes child entities from parents at runtime, the following commands must be used:
+```rust
+// removes the given children from the parent
+commands.entity(some_parent).remove_children(&[child1, child2]);
+// pushes the given children to the "end" of the parent's Children list
+commands.entity(some_parent).push_children(&[child3, child4]);
+// inserts the given children into the parent's Children list at the given index 
+commands.entity(some_parent).insert_children(1, &[child5]);
+``` 
+
+We've also added [`HierarchyEvent`], which makes it possible for developers to track changes in the hierarchy.
+
+There are still a couple of small holes to close, but staying on the "happy path" is much easier now:
+* removing only one of the components is possible (although heavily discouraged)
+* adding default values of only one of the components manually is still possible (although heavily discouraged)
+
+We're discussing ways to resolve this class of problem, such as [Archetype Rules / Invariants](https://github.com/bevyengine/bevy/issues/1481).
+
+[`Parent`]: https://docs.rs/bevy/0.8.0/bevy/hierarchy/struct.Parent.html
+[`Children`]: https://docs.rs/bevy/0.8.0/bevy/hierarchy/struct.Children.html
+[`HierarchyEvent`]: https://docs.rs/bevy/0.8.0/bevy/hierarchy/struct.HierarchyEvent.html
+
+## ECS Soundness / Correctness Improvements
+
+<div class="release-feature-authors">authors: @TheRawMeatball, @BoxyUwU, @SkiFire13, @jakobhellermann</div>
+
+Bevy ECS received a solid number of soundness and correctness bug fixes this release:
+
+* **Removed EntityMut::get_unchecked**: The only way to soundly use this API is already encapsulated within `EntityMut::get`. Therefore there was no reason to keep this unsafe API around.
+* **Fixed unsoundness with some `Or`/`AnyOf`/`Option` component access**: Previous Bevy versions allowed unsound versions of these queries. We now properly prevent those uses.
+* **Improve soundness of `CommandQueue`**: It is now sound to store Commands with padding or uninitialized bytes. A "use after move" case was also removed.
+* **Fix some memory leaks detected by miri**: Miri detected a case that leaks in our `BlobVec` drop impl. That is now fixed.
+* **Lint for missing SAFETY comments in bevy_ecs**: We now require safety comments for unsafe code blocks in `bevy_ecs`.
+
+As Bevy ECS matures, our bar for unsafe code blocks and soundness must also mature. Bevy ECS will probably never be 100% free of unsafe code blocks because we are modeling parallel data access that Rust cannot reason about without our help. But we are committed to removing as much unsafe code as we can and improving the quality and scope of our unsafe code.
```

Comment:
```suggestion
As Bevy ECS matures, our bar for unsafe code blocks and soundness must also mature. Bevy ECS will probably never be 100% free of unsafe code blocks because we are modeling parallel data access that Rust cannot reason about without our help. But we are committed to both removing as much unsafe code as we can and improving the quality of the unsafe code that remains.
```